src/Utils/GalleryUtils.ts

```diff
@@ -225,67 +226,88 @@ export function downloadItem(
   const name = data.name;
   useDialog.getState().showOkCancelModalDialog(
     `Download to ${useNotebook.getState().notebookFolderName}`,
-    `Download ${name} from gallery as a copy to your notebooks to run and/or edit the notebook.`,
+    NotebookUtil.getDownloadModalConent(name),
     "Download",
     async () => {
-      const clearInProgressMessage = logConsoleProgress(
-        `Downloading ${name} to ${useNotebook.getState().notebookFolderName}`
-      );
-      const startKey = traceStart(Action.NotebooksGalleryDownload, {
-        notebookId: data.id,
-        downloadCount: data.downloads,
-        isSample: data.isSample,
-      });
-
-      try {
-        const response = await junoClient.getNotebookContent(data.id);
-        if (!response.data) {
-          throw new Error(`Received HTTP ${response.status} when fetching ${data.name}`);
-        }
-
-        const notebook = JSON.parse(response.data) as Notebook;
-        removeNotebookViewerLink(notebook, data.newCellId);
-
-        if (!data.isSample) {
-          const metadata = notebook.metadata as { [name: string]: unknown };
-          metadata.untrusted = true;
-        }
-
-        await container.importAndOpenContent(data.name, JSON.stringify(notebook));
-        logConsoleInfo(`Successfully downloaded ${name} to My Notebooks`);
-
-        const increaseDownloadResponse = await junoClient.increaseNotebookDownloadCount(data.id);
-        if (increaseDownloadResponse.data) {
-          traceSuccess(
-            Action.NotebooksGalleryDownload,
-            { notebookId: data.id, downloadCount: increaseDownloadResponse.data.downloads, isSample: data.isSample },
-            startKey
+      if (NotebookUtil.isPhoenixEnabled()) {
+        await container.allocateContainer();
+      }
+      const notebookServerInfo = useNotebook.getState().notebookServerInfo;
+      if (notebookServerInfo && notebookServerInfo.notebookServerEndpoint !== undefined) {
+        downloadNotebookItem(name, data, junoClient, container, onComplete);
+      } else {
+        useDialog
+          .getState()
+          .showOkModalDialog(
+            "Failed to Connect",
+            "Failed to connect temporary environment, this could happen because of network issue please refresh and try again."
```

Comment:
What's the scenario when this error gets shown? Any other failure cases besides network connectivity?